airbyte-cdk/bulk/toolkits/load-gcs/src/test/kotlin/io/airbyte/cdk/load/file/gcs/GcsClientTest.kt

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.file.gcs
+
+import GcsBlob
+import GcsClient
+import com.google.cloud.storage.Blob
+import com.google.cloud.storage.BlobId
+import com.google.cloud.storage.BlobInfo
+import com.google.cloud.storage.CopyWriter
+import com.google.cloud.storage.Storage
+import io.airbyte.cdk.load.command.gcs.GcsClientConfiguration
+import io.airbyte.cdk.load.command.gcs.GcsHmacKeyConfiguration
+import io.airbyte.cdk.load.command.gcs.GcsRegion
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class GcsClientTest {
+    private lateinit var storage: Storage
+    private lateinit var config: GcsClientConfiguration
+    private lateinit var gcsClient: GcsClient
+
+    private val bucketName = "test-bucket"
+    private val bucketPath = "test-path"
+    private val region = GcsRegion.US_WEST1
+
+    private fun mockBlob(name: String): Blob {
+        val blob = mockk<Blob>()
+        every { blob.name } returns name
+        return blob
+    }
+
+    @BeforeEach
+    fun setup() {
+        storage = mockk()
+        val auth = GcsHmacKeyConfiguration("test-access-key", "test-secret-key")
+        config = GcsClientConfiguration(bucketName, bucketPath, auth, region)
+        gcsClient = GcsClient(storage, config)
+    }
+
+    @Test
+    fun `test list with empty prefix`() = runBlocking {
+        val blob1 = mockBlob("test-file1")
+        val blob2 = mockBlob("test-file2")
+        val blobs = mockk<Iterable<Blob>>()
```

Comment:
avoid mocking data structures. this should just be `listOf(blob1, blob2)`

(in fact, you could inline this as `every { iterateAll() } returns listOf(blob1, blob2)`)